src/components/data/PathListAutomation.js

```diff
@@ -0,0 +1,438 @@
+
+    let infoTypesCache = queryCache.getQueryData(INFO_TYPES_QUERY_KEY);
+
+    const { isFetching } = useQuery({
+        queryKey: INFO_TYPES_QUERY_KEY,
+        queryFn: getInfoTypes,
+        config: {
+            enabled: infoTypesQueryEnabled,
+            onSuccess: (resp) => {
+                setInfoTypeError(null);
+                setInfoTypes(resp.types);
+            },
+            staleTime: Infinity,
+            cacheTime: Infinity,
+            onError: setInfoTypeError,
+        },
+    });
+
+    const [createPathListFile, { status }] = useMutation(
+        ({ submission }) => pathListCreator(submission),
+        {
+            onSuccess: (data) => {
+                setCreatePathListError(null);
+                onCreatePathList(data.file.id, data.file.path);
+            },
+            onError: (error, { onError }) => {
+                let errMsg = null;
+                const err_code = getErrorCode(error);
+                switch (err_code) {
+                    case ERROR_CODES.ERR_EXISTS:
+                        errMsg = t("pathListExistError");
+                        onError(err_code);
+                        break;
+
+                    case ERROR_CODES.ERR_NOT_FOUND:
+                        errMsg = t("pathListNoMatchError");
+                        onError();
+                        break;
+
+                    default:
+                        errMsg = t("pathListCreateError");
+                        onError();
+                }
+                setCreatePathListError(error);
+                setErrorMsg(errMsg);
+            },
+        }
+    );
+
+    useEffect(() => {
+        if (!infoTypesCache || infoTypesCache.length === 0) {
+            setInfoTypesQueryEnabled(true);
+        } else {
+            if (infoTypes === null || infoTypes.length === 0) {
+                setInfoTypes(infoTypesCache.types);
+            }
+        }
+    }, [infoTypes, infoTypesCache]);
+
+    const handleToggle = (value) => () => {
+        const currentIndex = selectedInfoTypes.indexOf(value);
+        const newChecked = [...selectedInfoTypes];
+
+        if (currentIndex === -1) {
+            newChecked.push(value);
+        } else {
+            newChecked.splice(currentIndex, 1);
+        }
+
+        setSelectedInfoTypes(newChecked);
+    };
+
+    const handlePathListCreation = (values, actions) => {
+        const { selectedPaths, pattern, dest, fileName, foldersOnly } = values;
+
+        const submission = {
+            paths: selectedPaths,
+            dest: `${dest}/${fileName}`,
+            pattern,
+            foldersOnly: foldersOnly || false,
+            recursive: true,
+            requestedInfoType,
+            selectedInfoTypes,
+        };
+
+        const onError = (err_code) => {
+            if (err_code === ERROR_CODES.ERR_EXISTS) {
+                actions.setFieldError("dest", t("pathListExistError"));
+            }
+        };
+        if (status !== constants.LOADING) {
```

Comment:
Also, we can prevent dupe submission without disabling buttons by placing a guard just before  calling the mutation.